src/sw/sw.ts

```diff
@@ -0,0 +1,22 @@
+import {} from '.';
+declare var self: ServiceWorkerGlobalScope;
+
+import {NotificationPayload} from '../types/api';
+
+// This isn't a function to make typescript happy
+// intead its an "EventListener"
+const pushEventHandler = {
+  handleEvent: (event: PushEvent) => {
+    if (!event.data) {
+      return;
+    }
+
+    const data: NotificationPayload =
+        (JSON.parse(event.data.text()) as NotificationPayload);
+    event.waitUntil(self.registration.showNotification(
+        data['title'],
+        data,
+        ));
```

Comment:
Out of curiosity, are we required to show a notification when we receive a push event? Or can we simply cause the dashboard to refresh?

Reply:
We don't have to show a notification if there is a tab open and focused. In that scenario we can just refresh the pages (and we can refresh the pages regardless).